### ISSUE TITLE
[2.2.1] Instance-level Assistant Advanced Settings

### ISSUE THREAD

**Issue by rimacher** [Limekiller/moodle-block_openai_chat#72]
Hi,

I have two projects on my account, each with a different assistant. One project's key is set as the site-level default. I tried using the other project's key at the block level, but it still defaults to the site key.

**Comment by Limekiller** [comment]
There may be an issue with the standing of your OpenAI account. If you open your browser console to the network tab as you send the chat message, you should see a request to "completion.php." Can you send me the response to that request here so I can see what the issue is?

**Comment by rimacher** [comment]
Hi,

![Snipaste_2025-01-06_13-06-59](https://github.com/user-attachments/assets/c1a1a73d-35dc-42e4-bc7d-9ca8c2ebc0db)
I get 404 error.

The block settings show only the the site-level assistant and I am unable to select another one.

**Comment by Limekiller** [comment]
If you click on that request, what do you see in the response tab on the right?

**Comment by rimacher** [comment]
Hi,

I get "Exception - No assistant found with id"

    line 122 of /blocks/openai_chat/classes/completion/assistant.php: Exception thrown
    line 50 of /blocks/openai_chat/classes/completion/assistant.php: call to block_openai_chat\completion\assistant->run()
    line 93 of /blocks/openai_chat/api/completion.php: call to block_openai_chat\completion\assistant->create_completion()

**Comment by Limekiller** [comment]
The only way I was able to reproduce similar behavior is by

- Setting an API key as the site default
- Creating a block instance
- Editing the block instance and pasting in a different API key (note that the assistant list does not update right away)
- Saving the block and then attempting to use it

But of course this will cause an issue--the block settings need to be re-opened to select the correct assistant. When I re-open the block settings after saving the new API key, the list of assistants assigned to that project appear as expected. Saving it a second time with one of that project's assistants selected then correctly uses the selected assistant.

If your issue is that the assistant list is not updated when you re-open the block settings, I am not able to reproduce this behavior.

**Comment by rimacher** [comment]
Yes, my issue is that the assistant list is not updated when I re-open the block settings. Any idea what could have caused this?

**Comment by rimacher** [comment]
I managed to solve it by changing line 30 in file edit_form.php from:
$block_id = $this->_ajaxformdata["blockid"];
to:
$block_id = $this->block->instance->id;
